replication-api/src/main/java/com/connexta/replication/queue/Task.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.queue;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a task that was queued for later processing by a worker.
+ *
+ * <p>When a task is first queued, its total attempts counter will report <code>0</code> and its
+ * queued time and original queued time will be set to the same time (i.e. the current time when the
+ * task is queued via one of {@link Queue#offer(TaskInfo)}), {@link Queue#offer(TaskInfo, long,
+ * TimeUnit)}, or {@link Queue#put(TaskInfo)}. If the task fails and a decision is made to put it
+ * back into the queue via one of {@link #offerForRetry(PriorityLevel)}, {@link
```

Comment:
❓ Doesn't a Task only get removed from the queue when it is successfully performed? In which case we would never put the task back on the queue?